### PR TITLE
chore: post #3116 cleanup

### PR DESCRIPTION
**Pull request by justlevine**
What does this implement/fix? Explain your changes.
---------------------------------------------------

This PR does some code quality work on #3116 to improve readability / reduce complexity, and backfill some asserts.

Does this close any currently open issues?
------------------------------------------

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------

Any other comments?
-------------------
I don't think these warrant special changelog comments vs what's already there.

Where has this been tested?
---------------------------
**Operating System:** Ubuntu 20.04 (wsl2 + devilbox + php 8.1.15)

**WordPress Version:** 6.5.2

### CODE REVIEW

**Comment by justlevine** [comment]
@jasonbahl base branch is `develop` to get the ci to trigger, can be switched back to the release branch once its ready to merge.

**Comment by justlevine** [comment]
@jasonbahl my gut tells me theres some more cases we're missing in https://github.com/justlevine/wp-graphql/blob/6cd877cbb75f8175f5cafe37fa6bbc56f333740c/tests/wpunit/InterfaceTest.php#L547 but I'm to tired to figure out what and don't want to hold up the release. Consider this ready for review/merge.

**Review by jasonbahl — APPROVED** [review]

**Pull request merged**
